src/websockets/protocol.py

```diff
@@ -1001,18 +1020,18 @@ def writer_is_closing(self):
         """
         try:
             # Wait for the data transfer phase to complete.
-            if self.transfer_data_task is not None:
+            if hasattr(self, "transfer_data_task"):
```

Comment:
Is this the result of a typing issue or is the change unrelated to adding types?

If it is related to typing, have you considered `cast` in this case?

Reply:
Typing encouraged me to stop initializing attributes to `None` to avoid a plague of `Optional[...]` in type declarations.

It doesn't make a big difference in practice: if you use the attribute before it gets a valid value, you get "xxx has no attribute yyy" instead of "None has no attribute zzz" (when `xxx.yyy` = None).